org-roam-compat.el

```diff
@@ -95,6 +95,61 @@ recursion."
             (push (concat dir "/" file) files)))))
     (nconc result (nreverse files))))
 
+;;; Compatibility hacks and patches
+(advice-add #'org-id-add-location :around #'org-roam--handle-absent-org-id-locations-file-a)
```

Comment:
Going through the advice to `org-id-add-location`, because `org-id` related values can be altered through the runtime by the user (e.g. with dir-locals) and the related state can be configured in lazy loaded manner.